integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioEnabled.java

```diff
@@ -0,0 +1,402 @@
+  private static String managedServerPrefix = domainUid + "-" + MANAGED_SERVER_NAME_BASE;
+  private static int managedServerPort = 7100;
+  private static String finalPrimaryServerName = null;
+  private static String configMapName = "istio-configmap";
+  private static int replicaCount = 2;
+
+  private static LoggingFacade logger = null;
+
+  /**
+   * Install operator, create a custom image using model in image with model files
+   * and create a one cluster domain.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+
+    // get a new unique opNamespace
+    logger.info("Assign unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Assign unique namespace for Domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // Label the domain/operator namespace with istio-injection=enabled
+    Map<String, String> labelMap = new HashMap();
+    labelMap.put("istio-injection", "enabled");
+    assertDoesNotThrow(() -> addLabelsToNamespace(domainNamespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    List<String> appList = new ArrayList();
+    appList.add(SESSMIGR_APP_NAME);
+
+    // build the model file list
+    final List<String> modelList = Collections.singletonList(MODEL_DIR + "/" + SESSMIGR_MODEL_FILE);
+
+    // create image with model files
+    logger.info("Create image with model file and verify");
+    String miiImage = createMiiImageAndVerify(SESSMIGR_IMAGE_NAME, modelList, appList);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(miiImage);
+
+    // config the domain with Istio ingress with Istio gateway
+    String managedServerPrefix = domainUid + "-managed-server";
+    assertDoesNotThrow(() ->
+        configIstioModelInImageDomain(miiImage, domainNamespace, domainUid, managedServerPrefix),
+        "setup for istio based domain failed");
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  /**
+   * When Istio is enabled, test sends a HTTP request to set http session state(count number),
```

Comment:
When Istio is enabled --> In an istio enabled Environment

Reply:
changed